BinaryMatrixEngine.Tests/FullTurnTests.cs

```diff
@@ -0,0 +1,86 @@
+using System;
+using System.Collections.Immutable;
+using static BinaryMatrix.Engine.Axiom;
+using static BinaryMatrix.Engine.CellName;
+using static BinaryMatrix.Engine.Value;
+
+namespace BinaryMatrix.Engine.Tests;
+
+[TestClass]
+public class FullTurnTests {
+	[TestMethod]
+	public void MultipleComplexAttackers() {
+		StaticRNG rng = new(Array.Empty<int>());
+		TestPlayerActor a0a = new();
+		TestPlayerActor a1a = new();
+		TestPlayerActor a2a = new();
+		TestPlayerActor d0a = new();
+		Player a0 = new(PlayerRole.ATTACKER, 0, a0a);
+		Player a1 = new(PlayerRole.ATTACKER, 1, a1a);
+		Player a2 = new(PlayerRole.ATTACKER, 2, a2a);
+		Player d0 = new(PlayerRole.DEFENDER, 0, d0a);
+		using GameContext game = new(new[] { a0, a1, a2, d0 }, rng, TestGameHooks.CreateDefaultHooks());
+
+		game.board[L1].cards.AddRange(new[] { new Card(BREAK, DATA), new Card(SEVEN, CHOICE), new Card(WILD, VOID), new Card(THREE, KIN),  new Card(THREE, CHOICE) });
+		game.board[L2].cards.AddRange(new[] { new Card(SEVEN, FORM), new Card(FIVE, KIN), new Card(TWO, KIN), new Card(TRAP, VOID), new Card(TWO, CHAOS) });
+		game.board[L3].cards.AddRange(new[] { new Card(TWO, VOID), new Card(FIVE, KIN), new Card(TRAP, FORM) { revealed = true } });
+
+		game.board[A1].cards.AddRange(new[] { new Card(EIGHT, CHOICE), new Card(TRAP, CHAOS) });
+		game.board[A2].cards.AddRange(new[] { new Card(EIGHT, KIN), new Card(EIGHT, FORM) });
+
+		game.board[D1].cards.AddRange(new[] { new Card(FOUR, DATA) });
+		game.board[D2].cards.AddRange(new[] { new Card(FOUR, VOID) });
+
+		a1.Hand.Add(new Card(BREAK, CHOICE));
+		a2.Hand.Add(new Card(BREAK, KIN));
+		game.TurnCounter = 35;
+		a0a.currentAction = new ActionSet(ActionType.DRAW, 3);
+		a1a.currentAction = new ActionSet(ActionType.FACEUP_PLAY, new NativeCardSpecification(BREAK, null), 2);
+		a2a.currentAction = new ActionSet(ActionType.FACEUP_PLAY, new NativeCardSpecification(BREAK, null), 1);
+
+		PlayerRole? victor = game.Tick();
+		Assert.AreEqual(null, victor);
+		Assert.AreEqual(new [] {
```

Comment:
Missing state checks here. Testing the log is good, but the final state should be verified as well.